models/maespa/R/met2model.MAESPA.R

```diff
@@ -0,0 +1,265 @@
+##' @param end_date the end date of the data to be downloaded (will only use the year part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose should the function be very verbose
+##' 
+##'
+
+met2model.MAESPA <- function(in.path, in.prefix, outfolder, start_date, end_date, ..., overwrite=FALSE,verbose=FALSE){
+  
+  
+  library(PEcAn.utils)
+  print("START met2model.MAESPA")
+  start.date <- as.POSIXlt(start_date, tz = "GMT")
+  end.date<- as.POSIXlt(end_date, tz = "GMT")
+  out.file <- paste(in.prefix, strptime(start.date, "%Y-%m-%d"),strptime(end.date, "%Y-%m-%d"),"dat", sep=".")
+  out.file.full <- file.path(outfolder, out.file)
+  
+  results <- data.frame(file = out.file.full,
+                        host = fqdn(),
+                        mimetype ='text/csv',
+                        formatname = 'Sipnet.climna' ,
+                        startdate = start_date ,
+                        enddate = end_date,
+                        dbfile.name = out.file,
+                        stringsAsFactors = FALSE)
+  print("internal results")
+  print(results)
+  
+  if (file.exists(out.file.full) && !overwrite) {
+    logger.debug("File '", out.file.full, "' already exists, skipping to next file.")
+    return(invisible(results))
+  }
+  
+  
+  require(ncdf4)
+  require(lubridate)
+  require(PEcAn.data.atmosphere)
+  
+  ## check to see if the outfolder is defined, if not create directory for output
+  if(!file.exists(outfolder)){
+    dir.create(outfolder)
+  }
+  
+  out <- NULL
+  
+  # get start/end year since inputs are specified on year basis
+  start_year <- year(start.date)
+  end_year <- year(end.date)
+  
+  ## loop over files
+  for(year in start_year:end_year){
+    print(year)
+    
+    old.file <- file.path(in.path, paste(in.prefix, year, "nc", sep="."))
+    
+    if(file.exists(old.file)){
+      ## open netcdf
+      nc <- nc_open(old.file)
+      ## convert time to seconds
+      sec   <- nc$dim$time$vals  
+      sec = udunits2::ud.convert(sec,unlist(strsplit(nc$dim$time$units," "))[1],"seconds")
+      
+      ifelse(leap_year(year)==TRUE,
+             dt <- (366*24*60*60)/length(sec), #leap year
+             dt <- (365*24*60*60)/length(sec)) #non-leap year
+      tstep = round(86400/dt)
+      dt = 86400/tstep
+      
+      #Check wich variabales are available and which are not
+      
+      ## extract variables
+      lat  <- ncvar_get(nc,"latitude")
+      lon  <- ncvar_get(nc,"longitude")       
+      #RAD <-  ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") #W m-2
+      PAR <-   try(ncvar_get(nc,"surface_downwelling_photosynthetic_photon_flux_in_air")) #mol m-2 s-1
+      TAIR <-  ncvar_get(nc,"air_temperature") #K
+      #TSOIL <-  try(ncvar_get(nc,"soil_temperature")) #K
+      #RHperc <-  try(ncvar_get(nc,"relative_humidity_percentage")) 
+      `RH%` <- try(ncvar_get(nc,"relative_humidity"))
```

Comment:
MAESPA expects "RH" as a fraction and "RH%" as a percentage and percentage is MSTMIP standard so I use RH%.